Allow use of a shared cookbook path

chefsalim opened the pull request:
Enable chef-acceptance to pull in a shared cookbook path.
This allows us to have common infrastructure used between multiple suites.

@sersut @patrick-wright @jkeiser

sersut commented:
I'm :-1: on this one. 

IMO if sharing code this way is super important, let's write a card to get `berks` support in and put it at the top of the queue. Until we do that card we can just copy paste things for now.

patrick-wright commented:
I concur with Serdar.  We should be managing cookbook dependencies with Berks or even better, policyfiles.

chefsalim commented:
Per offline discussion:
1. Added comment to indicate this change is only for internal consumption.
2. Added an integration test.
3. Card for extending support for external usage - https://chefio.atlassian.net/secure/RapidBoard.jspa?rapidView=70&projectKey=ECO&view=detail&selectedIssue=ECO-278

sersut commented:
:+1:
